dm/master/server.go

```diff
@@ -3218,3 +3218,72 @@ func genValidationWorkerErrorResp(req *workerrpc.Request, err error, logMsg, wor
 		return nil
 	}
 }
+
+func (s *Server) UpdateValidation(ctx context.Context, req *pb.UpdateValidationRequest) (*pb.UpdateValidationResponse, error) {
+	var (
+		resp2 *pb.UpdateValidationResponse
+		err   error
+	)
+	shouldRet := s.sharedLogic(ctx, req, &resp2, &err)
+	if shouldRet {
+		return resp2, err
+	}
+	sources := req.Sources
+	if len(sources) == 0 {
+		sources = s.getTaskSourceNameList(req.TaskName)
+		if len(sources) == 0 {
+			return &pb.UpdateValidationResponse{
+				Result: false,
+				Msg:    fmt.Sprintf("task %s has no source or not exist, please check the task name and status", req.TaskName),
+			}, nil
+		}
+	}
+
+	workerReq := workerrpc.Request{
+		Type: workerrpc.CmdUpdateValidation,
+		UpdateValidation: &pb.UpdateValidationWorkerRequest{
+			TaskName:   req.TaskName,
+			BinlogPos:  req.BinlogPos,
+			BinlogGTID: req.BinlogGTID,
```

Comment:
How about adding a check for args that `BinlogPos` is empty and `EnableGTID` is false or `BinlogGTID` is empty and `EnableGTID` is true?

Reply:
I will fix it in next PR